rero_ils/modules/acq_orders/api.py

```diff
@@ -184,6 +184,9 @@ def reasons_not_to_delete(self):
         """Get reasons not to delete record."""
         cannot_delete = {}
         links = self.get_links_to_me()
+        # allow deletion of order and its line when its status is still PENDING
+        if self.status == AcqOrderStatus.PENDING:
+            links.pop('acq_order_lines', None)
```

Comment:
As mentionned, AcqOrderLine is never a reason why not to delete. But status is a valid reason :
```
links.pop('acq_order_lines', None)
if self.status != AcqOrderStatus.PENDING:
    cannot_delete['others'] = {
        _(f'Order status is {self.status}'): true
    }
```